File: server/src/models/Visit.js
const mongoose = require('mongoose');

const visitSchema = new mongoose.Schema({
  patientId: { type: mongoose.Schema.Types.ObjectId, ref: 'Patient', required: true, index: true },
  doctorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Doctor', required: true },
  visitDate: { type: Date, required: true },
  hba1c: { type: Number, required: true },
  creatinine: { type: Number, required: true },
  bloodUrea: { type: Number, required: true },
  eGFR: { type: Number, required: true },
  systolicBP: { type: Number, required: true },
  diastolicBP: { type: Number, required: true },
  urineACR: { type: Number, required: true },
  ckdStageAtVisit: { type: Number },
  riskScoreAtVisit: { type: Number },
  riskLevelAtVisit: { type: String },
  notes: { type: String, default: '' }
}, { timestamps: true });

module.exports = mongoose.model('Visit', visitSchema);

File: server/src/main/java/com/nephroalert/entity/Visit.java
package com.nephroalert.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "visits")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Visit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long patientId;

    @Column(nullable = false)
    private Long doctorId;

    @Column(nullable = false)
    private LocalDate visitDate;

    @Column(nullable = false)
    private Double hba1c;

    @Column(nullable = false)
    private Double creatinine;

    @Column(nullable = false)
    private Double bloodUrea;

    @Column(nullable = false)
    private Integer eGFR;

    @Column(nullable = false)
    private Integer systolicBP;

    @Column(nullable = false)
    private Integer diastolicBP;

    @Column(nullable = false)
    private Double urineACR;

    private Integer ckdStageAtVisit;

    private Integer riskScoreAtVisit;

    private String riskLevelAtVisit;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
